lib/utils/debouncer.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+class Debouncer {
+  final Duration _duration;
+  final ValueNotifier<bool> _debounceActiveNotifier = ValueNotifier(false);
+  Timer _debounceTimer;
+
+  Debouncer(this._duration);
+
+  void run(Future<void> Function() fn) {
+    if (isActive()) {
+      _debounceTimer.cancel();
+    }
+    _debounceTimer = Timer(_duration, () async {
+      await fn();
+      _debounceActiveNotifier.value = false;
+    });
+    _debounceActiveNotifier.value = true;
+  }
+
+  void cancelDebounce() {
```

Comment:
Can simply be `cancel`?

Reply:
I renamed it to cancelDebounce() because I found myself confused once with the cancel() method in the Timer class and this method, do you think this helps?